samples/star/src/main/kotlin/com/slack/circuit/star/petlist/PetList.kt

```diff
@@ -146,23 +157,27 @@ constructor(
 ) : Presenter<PetListScreen.State> {
   @Composable
   override fun present(): PetListScreen.State {
+    var isRefreshing by remember { mutableStateOf(false) }
     val animalState by
-      produceRetainedState<List<PetListAnimal>?>(null) {
-        val animals = petRepo.getAnimals()
+      produceRetainedState<List<PetListAnimal>?>(null, isRefreshing) {
+        val animals = petRepo.getAnimals(isRefreshing)
+        isRefreshing = false
         value = animals.map { it.toPetListAnimal() }
       }
 
-    return remember(screen, animalState) {
+    return remember(screen, animalState, isRefreshing) {
```

Comment:
I know it predates your PR but this might have had something to do with the issues you were seeing - we shouldn't be remembering these returns I think? @kierse and I saw some issues before, and we should just be returning the final result directly

Reply:
hmm...I thought I had to include the `isRefreshing` here in order for the state to pick up the current value of that boolean. I guess now that I pass it into `produceRetainedState` it's not necessary here anymore? Am I understanding that correctly.

Reply:
Correct. I've already gone and removed it on main if you pull latest

Reply:
Addressed in [d76cb7b](https://github.com/slackhq/circuit/pull/142/commits/d76cb7bf88060db12e6669cba221681c9ef8815e)

Reply:
you misunderstood I think, I meant remove this `remember` entirely and just return the state directly